keycloak/keycloak-admin-ui
# User fed: Add LDAP provider and tests

## Motivation
https://issues.redhat.com/browse/APPDUX-806
https://issues.redhat.com/browse/APPDUX-847

## Brief Description
Decision in the 2/2/21 keycloak status meeting was to move forward with functionality to add new providers directly using the 'Add new provider' dropdown in the User Federation settings view similar to the way it was done in the old admin console UI, instead of via the wizards, at least for now.

Note that this PR also covers creating a new LDAP provider via the user fed empty state page as well, by clicking the 'Add LDAP providers' card.

Also included in this PR is a suite of cypress tests for the LDAP user federation provider as a whole, which includes tests for:

- Creating an LDAP provider from empty state
- Creating an LDAP provider from the card view
- Changing an existing LDAP provider and saving (update)
- Changing an existing LDAP provider and canceling (cancel)
- Disabling an LDAP provider
- Enabling a previously-disabled LDAP provider
- Deleting an LDAP provider using the card menu
- Deleting an LDAP provider using the settings view Action menu

## Verification Steps

1. Go to user federation with no LDAP providers configured.
2. Click the 'Add LDAP providers' card and verify that you can successfully create a new LDAP provider using the form that follows.
3. With at least one LDAP provider configured, verify that you can successfully create a new LDAP provider using the Add new provider > LDAP command.
4. Bonus points: run yarn start:cypress and verify that the LDAP provider tests all pass.

## Checklist:


## Additional Notes
The follow-up issues that were filed for the kerberos provider and tests PR (https://github.com/keycloak/keycloak-admin-ui/pull/371) also apply to this PR, and will be implemented as part of those fixes:
https://github.com/keycloak/keycloak-admin-ui/issues/374 (don't save form unless dirty)
https://github.com/keycloak/keycloak-admin-ui/issues/375 (don't allow multiple saves during create)
https://github.com/keycloak/keycloak-admin-ui/issues/376 (change tests to allow for existing providers)
